docs/polkadot/networking/3-attestation.md

```diff
@@ -1,54 +1,268 @@
-- **Valid:**
-    - Members
-        - Candidate Hash
-        - Validator public key
-        - Signature
-        - Relay Chain Parent Hash
-    - This is an attestation on the candidate with given hash, that indicates the candidate is both valid and has block data available. May only be sent to peers who are aware of the `Candidate`.
-- **Invalid:**
-    - Members
-        - Candidate Hash
-        - Validator public key
-        - Signature
-        - Relay Chain Parent Hash
-    - This is an attestation against the candidate with given hash. May only be sent to peers who are aware of the `Candidate`.
+Let T_B represent the average time it takes for a distributor to transfer all pieces of a block, i.e. a single piece takes roughly T_B/N. This is only the bandwidth and not latency, so only if T_B/N is negligible then the successful direct transfer of a single piece including acknowledge takes about 1 RTT.
 
-### Direct Routing: sending/ receiving erasure coded pieces
+Let T_b be T_B divided by the number of validators per parachain, i.e. T_b = T_B/(N/C). This represents the average time it takes for a distributor to transfer the pieces of a block if this responsibility was shared equally across all validators of a parachain.
+
+Let T_L represent the standard deviation of the distribution of when validators begin this distribution process for each period.
+
+In practise based on some crude real-world measurements, we guess that T_l is about 150ms, T_b is about 3s, and T_L is about 750ms. In other words, T_b ~= 20 * T_l ~= 4 * T_L.
+
+## High-level requirements
+
+Ensure that ceil(N/3) of the pieces of all C blocks remains available and retrievable for a reasonable amount of time, across the N validators.
+
+## Design considerations
+
+### General aims
+
+A1: Reduced redundancy & latency: pieces should (mostly) only be sent to the nodes that need it, i.e. its main storer, and the secondary checkers.
+
+A2: Constrain the use of shared OS resources, so the protocol does not interfere with other programs running at the same site.
+
+A3: Performance degrades reasonably under load or imperfect network conditions.
+
+A4: TODO: Protocols should not be easily spammable, or the spammers should be easily identifiable and punishable.
+
+### Special considerations
+
+S1: A small fraction of nodes may have poor reachability, and need to communicate via a proxy. Otherwise, we can assume the graph of communication links is mostly-connected via the public internet.
+
+This is an optional consideration; we believe we do not need to consider it in the initial implementation, and so our solution for this is described in a separate section in this document. Parts of the main proposal that will need to be changed or extended for this, are marked [TBX S1 #($ref)], TBX standing for "to be extended".
+
+## Protocol overview
+
+We rely on an underlying gossip network that allows us to broadcast various metadata to everyone, namely:
```

Comment:
Do I understand correctly that for each period each validator gossips one metadata message? Thus with 1_000 validators there would be 1_000 meta data gossip messages per period?